snapcraft/plugins/catkin_tools.py

```diff
@@ -0,0 +1,124 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The catkin_tools build plugin is useful for building ROS parts.
+
+This plugin relies on the catkin plugin as the catkin plugin performs
+configuration that allows catkin_tools to run.  This plugin uses the
+same keywords and configurations as the catkin plugin, the difference
+is the installation of and using catkin_tools to build.
+"""
+
+import os
+import logging
+
+import snapcraft.plugins.catkin
+from snapcraft.plugins.catkin import (
+    Compilers,
+)
+
+import snapcraft
+
+logger = logging.getLogger(__name__)
+
+
+class CatkinToolsPlugin(snapcraft.plugins.catkin.CatkinPlugin):
+    def __init__(self, name, options, project):
+        super().__init__(name, options, project)
+        self.build_packages.extend(['python-catkin-tools'])
+
+    def _prepare_build(self):
+        super()._prepare_build()
+
+        self._clean_catkin()
+
+        self._add_catkin_profile()
+
+        self._configure_catkin_profile()
+
+    def _clean_catkin(self):
+        # Clean to prevent conflicts with externally built packages.
+        catkincmd = ['catkin', 'clean', '-y']
+
+        self._run_in_bash(catkincmd)
+
+    def _add_catkin_profile(self):
+        # Overwrite the default catkin profile to ensure builds
+        # aren't affected by profile changes.
+        catkincmd = ['catkin', 'profile', 'add', '-f', 'default']
+
+        self._run_in_bash(catkincmd)
+
+    def _configure_catkin_profile(self):
+        # Use catkin config to set all configurations before running build.
+        catkincmd = ['catkin', 'config']
+
+        # Use the newly created default profile.
+        catkincmd.extend(['--profile', 'default'])
```

Comment:
Why not set this profile to be active in `_add_catkin_profile()`? It would save your needing to explicitly use it in every subsequent command, no?

Reply:
Because there was only one subsequent command, I'm good to change that.

Reply:
No need, just a thought.